### PR TITLE
amp-action-macro extension skeleton

### PR DESCRIPTION
**Pull request by alabiaga**
Add the directory structure, experiment, and minimal viable changes for amp-action-macro component. This PR excludes the actual implementation to reduce the size of the PR and improve ease of reviewing.

Please see #19494

### CODE REVIEW

File: extensions/amp-action-macro/0.1/amp-action-macro.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  /** @override */
+  getLayoutPriority() {
+    // Loads after other content.
+    return LayoutPriority.METADATA;
+  }
+
+  /** @override */
+  isAlwaysFixed() {
+    return true;
```
Comment (aghassemi) [inline extensions/amp-action-macro/0.1/amp-action-macro.js]: This shouldn't be needed. Not sure why `amp-analytics` and `amp-state` override it. This is meant for components like `amp-sticky-ad1 and `amp-user-notifications` which are always `position:fixed`

File: extensions/amp-action-macro/amp-action-macro.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    <td width="40%"><strong>Description</strong></td>
+    <td>Creates reusable actions.</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Availability</strong></td>
+    <td><a href="https://www.ampproject.org/docs/reference/experimental.html">Experimental</a></td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Required Script</strong></td>
+    <td><code>&lt;script async custom-element="amp-viz-vega" src="https://cdn.ampproject.org/v0/amp-action-macro-0.1.js">&lt;/script></code></td>
```
Comment (aghassemi) [inline extensions/amp-action-macro/amp-action-macro.md]: custom-element is wrong

File: extensions/amp-action-macro/amp-action-macro.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  </tr>
+  <tr>
+    <td width="40%"><strong>Examples</strong></td>
+    <td><a href="https://github.com/ampproject/amphtml/blob/master/examples/amp-action-macro.amp.html">amp-action-macro.amp.html</a></td>
+  </tr>
+</table>
+ [TOC]
+ ## Overview?
+The amp-action-macro component allows for the creation of reusable actions.
```
Comment (aghassemi) [inline extensions/amp-action-macro/amp-action-macro.md]: The `amp-action-macro` component

File: extensions/amp-action-macro/0.1/amp-action-macro.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {LayoutPriority} from '../../../src/layout';
+
+/**
+* The <amp-action-macro> element is used to define a reusable action macro.
+*/
+export class AmpActionMacro extends AMP.BaseElement {
```
Comment (aghassemi) [inline extensions/amp-action-macro/0.1/amp-action-macro.js]: you probably wanna add the buildCallback and assert the `amp-action-micro` experiment

**Comment by aghassemi** [comment]
@alabiaga @choumx Awesome feature!

Is there a design to look at? I am particularly interested in whether we can make the `action` part as parametrizable as possible. It would help with a lot of cases currently impossible to do, specially if parameter values can come from `amp-state`. Super flexible case where `action` can be a `bind expression` would be absolutely amazing since it would additionally provide conditional actions which has been requested in the past.

e.g. Having the following would be amazing!

```
<amp-action-macro id="toggleCurrentAnimation"
    action-expression="myState.curAnimation.Id + '.' + (myState.curAnimation.running ? 'pause' : 'start')  ">
</amp-action-macro>
```

**Comment by alabiaga** [comment]
@aghassemi there is no design. I was planning on just implementing what was necessary to fulfill the needs of the following FR: #19189, where the action expressions are not dynamic. But I see no reason why we couldn't extend it to do much more, perhaps in another iteration or in my implementation. I'll need to assess the scope of the work and will capture that in a doc. Thanks

**Comment by choumx** [comment]
We should discuss this in the next design review. :)

File: tools/experiments/experiments.js [hunk 5]
```diff
@@ -147,6 +147,12 @@ const EXPERIMENTS = [
     spec: 'https://github.com/ampproject/amphtml/issues/6106',
     cleanupIssue: 'https://github.com/ampproject/amphtml/pull/6351',
   },
+  {
+    id: 'amp-action-micro',
```
Comment (choumx) [inline tools/experiments/experiments.js]: macro

Comment (alabiaga) [reply tools/experiments/experiments.js]: doh. #19591 thanks

**Pull request merged**